modules/core/src/bitcoin.ts

```diff
@@ -42,6 +42,32 @@ function getKey(network?: bitcoin.Network): bitcoin.ECPair {
 
 bitcoin.HDNode.prototype.getKey = getKey;
 
+/**
+ * Given a key and a path, derive the child key.
+ * @param {bitcoin.HDNode} userKey
+ * @param {string} path
+ * @returns {bitcoin.HDNode}
+ */
+export function deriveKeyByPath(userKey: bitcoin.HDNode, path: string): bitcoin.HDNode {
+  let key = userKey;
+  let splitPath = path.split('/');
+  // if a key path starts with "m", it is the path for a master node. derivePath() is used specifically for
+  // deriving master node and we can call it directly.
+  if (splitPath[0] === 'm') {
+    key = userKey.derivePath(path);
+  } else {
+    // if the path does not start with "m", it typically looks like "/x/y/...", and the splitPath
+    // would look like ['', 'x', 'y',...], and we need to get ride of the empty string at index 0.
+    // Then we continue deriving the child by calling derive() on the new child at each subsequent level.
+    splitPath = splitPath.slice(1);
+    for (const p of splitPath) {
+      const index = parseInt(p, 10);
```

Comment:
we should usually check `parseInt` result with `isNaN` but in this case we should go even further

sadly `parseInt` is very lenient 

```
parseInt(`1'`) === 1
parseInt(`1x`) === 1
```

Let's be extra strict here and throw and throw an error if `index.toString() !== p`